packages/thirdweb/src/chainsaw/formatter.ts

```diff
@@ -0,0 +1,123 @@
+import { BigNumber } from "ethers";
+import { type GetBlockReturnType, formatBlock, formatTransaction } from "viem";
+import type { Hex } from "../utils/encoding/hex.js";
+import { parseNFT } from "../utils/nft/parseNft.js";
+import type {
+  Block,
+  ChainsawTransactions,
+  NFTs,
+  NFTsData,
+  Transaction,
+  Transactions,
+} from "./types.js";
+
+export const formatChainsawBlock = (
+  block?: Block,
+): GetBlockReturnType<undefined, false> | undefined => {
+  if (!block) return;
+  return formatBlock({
+    ...block,
+    baseFeePerGas:
+      `0x${BigNumber.from(block.baseFeePerGas).toHexString()}` as Hex,
+    difficulty: `0x${BigNumber.from(block.difficulty).toHexString()}` as Hex,
+    gasLimit: `0x${BigNumber.from(block.gasLimit).toHexString()}` as Hex,
+    gasUsed: `0x${BigNumber.from(block.gasUsed).toHexString()}` as Hex,
+  }) as GetBlockReturnType<undefined, false>;
+};
+
+export const formatChainsawNFTs = (nfts?: NFTsData): NFTs => {
+  if (!nfts) return [];
+  return nfts.map((nft) => {
+    const options =
+      nft.type === "ERC1155"
+        ? {
+            tokenId: BigInt(nft.tokenId),
+            owner: nft.ownerAddress || null,
+            tokenUri: nft.uri,
+            type: "ERC1155" as const,
+            supply: BigInt(nft.balance || 0),
```

Comment:
Supply = balance?

Reply:
I assume it's the collection supply? Then yes, it does not equal balance. Changed it to 0 as we don't really have that information about the collection
[8b435ae](https://github.com/thirdweb-dev/js/pull/3741/commits/8b435ae2f660e86767d07c0a44c8d68b687f7d9a)